internal/app/machined/pkg/system/services/kubelet.go

```diff
@@ -120,6 +120,7 @@ func (k *Kubelet) Runner(r runtime.Runtime) (runner.Runner, error) {
 	mounts := []specs.Mount{
 		{Type: "bind", Destination: "/dev", Source: "/dev", Options: []string{"rbind", "rshared", "rw"}},
 		{Type: "sysfs", Destination: "/sys", Source: "/sys", Options: []string{"bind", "ro"}},
+		{Type: "securityfs", Destination: "/sys/kernel/security", Source: "/sys/kernel/security", Options: []string{"bind", "ro"}},
```

Comment:
should it need to be just "ro"? so kubelet only looks into it, but never configures anything?

do we want it to be backported or part of some future work?

Reply:
`bind` should be needed because that will propagate changes from the host down into the container (if anything does change over time I am unsure of).

Reply:
I don't think it _needs_ to be backported but if it wouldn't hurt then it could be useful in 1.7. I can check with the user.

Reply:
Only `ro` is technically needed for AppArmor since `kubelet` is checking if AppArmor is enabled via opencontainer apparmor lib: https://github.com/opencontainers/runc/blob/main/libcontainer/apparmor/apparmor_linux.go#L18

Which check if directory `/sys/kernel/security/apparmor` exists and a read on `/sys/module/apparmor/parameters/enabled` returns `Y`.
Both which should have their final value set by the time `kubelet` starts.
The part I'm not sure how `bind` handles is that `/sys/kernel/security/`is one of these virtual filesystems where this `securityfs` is a sort of interface to LSM modules in the kernel.

Reply:
I think `bind` should be fine, it just exposes the filesystem at another mountpoint, but without bind we do a completely fresh mount.

Reply:
So do we want it?

Reply:
yep, I think `bind` is better, but we also need to handle the case when the filesystem is not mounted on the host (if running in container mode). I believe in container mode we don't mount it explicitly (???) and rely on the container runtime to provide it or not... just to make sure we don't break Talos in containers if securityfs is not mounted on the container runtime host. I don't know if it's even possible though, just as a precaution :)

Reply:
yep, we don't mount securityfs in container mode: 

https://github.com/siderolabs/talos/blob/ccdb4c8b10450aa7fb6c32b0559bda73746a03ed/internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer.go#L88-L97

But as we do bind mount, I believe our code will just mount an empty directory into the kubelet if the host doesn't provide securityfs, so bind >> re-mount :)